plugins/modules/cloud/misc/proxmox_snap.py

```diff
@@ -0,0 +1,289 @@
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: community.general.proxmox_snap
+short_description: Snapshot management of instances in Proxmox VE cluster.
+description:
+  - Allows you to create/delete snapshots from instances in Proxmox VE cluster.
+options:
+  api_host:
+    description:
+      - The host of the Proxmox VE cluster.
+    required: true
+    type: str
+  api_user:
+    description:
+      - The user to authenticate with.
+    required: true
+    type: str
+  api_password:
+    description:
+      - The password to authenticate with.
+      - You can use PROXMOX_PASSWORD environment variable.
+    type: str
+  hostname:
+    description:
+      - The instance name.
+    type: str
+  vmid:
+    description:
+      - The instance id.
+      - If not set, will be fetched from PromoxAPI based on the hostname.
+    type: str
+  validate_certs:
+    description:
+      - Enable / disable https certificate verification.
+    type: bool
+    default: no
+  state:
+    description:
+     - Indicate desired state of the instance snapshot.
+    choices: ['present', 'absent']
+    default: present
+    type: str
+  force:
+    description:
+      - For removal from config file, even if removing disk snapshot fails.
+    default: no
+    type: bool
+  vmstate:
+    description:
+      - Snapshot includes RAM.
+    default: no
+    type: bool
+  description:
+    description:
+      - Specify the description for the snapshot. Only used on the configuration web interface.
+      - This is saved as a comment inside the configuration file.
+    type: str
+  timeout:
+    description:
+      - Timeout for operations.
+    default: 30
+    type: int
+  snapname:
+    description:
+      - Name of the snapshot that has to be created.
+    default: 'ansible_snap'
+    type: str
+
+notes:
+  - Requires proxmoxer and requests modules on host. This modules can be installed with pip.
+requirements: [ "proxmoxer", "python >= 2.7", "requests" ]
+author: Jeffrey van Pelt (@Thulium-Drake)
+'''
+
+EXAMPLES = r'''
+- name: Create new container snapshot
+  community.general.proxmox_snap:
+    api_user: root@pam
+    api_password: 1q2w3e
+    api_host: node1
+    vmid: 100
+    state: present
+    snapname: pre-updates
+
+- name: Remove container snapshot
+  community.general.proxmox_snap:
+    api_user: root@pam
+    api_password: 1q2w3e
+    api_host: node1
+    vmid: 100
+    state: absent
+    snapname: pre-updates
+
+- name: Make snapshots of multiple machines
+  hosts: all
+  gather_facts: false
+  tasks:
+    - name: make snapshots using the vmid
+      community.general.proxmox_snap:
+        api_user: root@pam
+        api_password: 1q2w3e
+        api_host: node1
+        vmid: "{{ proxmox_vmid }}"
+        state: present
+        snapname: test
+        vmstate: true
+      delegate_to: localhost
+    - name: remove snapshots using the inventory hostname
```

Comment:
```suggestion
      delegate_to: localhost
  
    - name: Remove snapshots using the inventory hostname
```
and is `delegated_to: localhost` really needed here?

Reply:
Not really required, but as this module has an external dependency on the ```proxmoxer``` library, it makes sense to install that only on the Ansible controller. It should not be run on the 'target' host itself.